presto-testing/src/main/java/io/prestosql/testing/datatype/DataType.java

```diff
@@ -180,14 +185,38 @@
 
     public static DataType<LocalTime> timeDataType(int precision)
     {
-        String pattern = "'TIME '''HH:mm:ss" + (precision == 0 ? "" : ("." + "S".repeat(precision))) + "''";
+        // This code does not support precision > 9, as java.time classes support precision up to nanoseconds
+        checkArgument(precision >= 0 && precision <= 9, "Unsupported precision: %s", precision);
+        String pattern = "'TIME '''HH:mm:ss" +
+                (precision == 0 ? "" : ("." + "S".repeat(precision)))
+                + "''";
+
         return dataType(
                 format("time(%s)", precision),
                 createTimeType(precision),
                 DateTimeFormatter.ofPattern(pattern)::format,
                 identity());
     }
 
+    public static DataType<OffsetTime> timeWithTimeZoneDataType(int precision)
+    {
+        // This code does not support precision > 9, as java.time classes support precision up to nanoseconds
+        checkArgument(precision >= 0 && precision <= 9, "Unsupported precision: %s", precision);
+
+        return dataType(
+                format("time(%s) with time zone", precision),
+                createTimeWithTimeZoneType(precision),
+                new DateTimeFormatterBuilder()
+                        .appendPattern("'TIME '''HH:mm:ss" + (precision == 0 ? "" : ("." + "S".repeat(precision))))
+                        .appendOffset("+HH:mm", "+00:00")
+                        .appendPattern("''")
+                        .toFormatter()::format);
```

Comment:
Pushed a change here. This looks horrible. We're fighting against our own convenience. I hope https://github.com/prestosql/presto/pull/6345 will eliminate all this.